api/datadoghq/v2alpha1/datadogagent_validation.go

```diff
@@ -0,0 +1,18 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package v2alpha1
+
+import "fmt"
+
+// ValidateDatadogAgent is used to check if a DatadogAgent is valid
+func ValidateDatadogAgent(dda *DatadogAgent) error {
```

Comment:
Not for this PR, but we should get rid of this validation in code and instead use kubebuilder markers: something like `+kubebuilder:validation:Required` or `+kubebuilder:validation:MinLength` https://book.kubebuilder.io/reference/markers/crd-validation